CMakeLists.txt

```diff
@@ -0,0 +1,857 @@
+cmake_minimum_required(VERSION 3.0)
+
+# Based on configure.ac
+
+project(gperftools VERSION 2.6.1 LANGUAGES C CXX)
+
+set(tcmalloc_so_version 8.5.4)
+set(profiler_so_version 4.14.4)
+
+set(default_build_cpu_profiler ON)
+set(default_build_heap_profiler ON)
+set(default_build_heap_checker ON)
+set(default_build_debugalloc ON)
+set(default_build_minimal OFF)
+
+set(default_tcmalloc_alignment 16)
+set(need_nanosleep ON)
+
+set(host string(TOLOWER "${CMAKE_SYSTEM_NAME}"))
+
+if(MINGW)
+  set(default_build_minimal ON)
+  set(default_build_debugalloc OFF)
+  set(need_nanosleep OFF)
+elseif(CYGWIN)
+  set(default_build_heap_checker OFF)
+  set(default_build_cpu_profiler OFF)
+elseif(host MATCHES "freebsd")
+  set(default_build_heap_checker OFF)
+elseif(APPLE)
+  set(default_build_heap_checker OFF)
+endif()
+
+include(CheckCCompilerFlag)
+include(CheckCSourceCompiles)
+include(CheckCXXSourceCompiles)
+include(CheckFunctionExists)
+include(CheckIncludeFile)
+include(CheckLibraryExists)
+include(CheckSymbolExists)
+include(CheckTypeSize)
+include(CheckVariableExists)
+include(CMakeDependentOption)
+include(CTest)
```

Comment:
AFAIK `CTest` is included automatically. `add_test()` is a builtin - unless you're using it for something else?

Reply:
`BUILD_TESTING` is only defined when including this module. See https://cmake.org/cmake/help/latest/module/CTest.html.

Reply:
That's by design. `BUILD_TESTING` is off by default, for good reason. Most users define `-DBUILD_TESTING=On` in order to automatically build tests. Then the workflow becomes simply `make` and `ctest` to run the entire project's suite of tests.